robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -275,29 +268,30 @@ def associate(self, original_entity):
         else:
             return original_entity
 
-    def distance(self, e1, e2):
+    @staticmethod
+    def distance(e1, e2):
         """ Computes the distance between two entities """
         return e1.distance_to_3d(e2._pose.p)
 
 
 class ResetOnFailure(smach.StateMachine):
     """ Class to reset the robot after a grab has failed """
-    def __init__(self, robot):
+    def __init__(self, robot, arm):
         """ Constructor
         :param robot: robot object
         """
-        smach.StateMachine.__init__(self, outcomes=['done'], input_keys=['arm'])
+        smach.StateMachine.__init__(self, outcomes=['done'])
 
         self._robot = robot
+        self.arm_designator = arm
 
-    def execute(self, userdata):
+    def execute(self, userdata=None):
         """ Execute hook """
-        if hasattr(userdata, 'arm'):
-            arm = self._robot.get_arm(required_gripper_types=arms.GripperTypes.GRASPING,
-                                      required_arm_name=userdata.arm.side)
-        else:
-            arm = None
-        self._robot.torso.reset()  # Move up to make resetting of the arm safer
+        arm = self.arm_designator.resolve()
+        arm.reset()
+
+        # ToDo: move up... This is AMIGO specific
```

Comment:
Isn't this an easy fix? If so please do it now!

Reply:
Easy hack...